### PR TITLE
Add docs for shardGrepEnabled

### PR DESCRIPTION
**Pull request by FriggaHel**
### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

File: docs/web-apps/automated-testing/playwright/yaml.md [hunk 1]
```diff
@@ -714,6 +714,34 @@ The `numShards` and `shard` properties are mutually exclusive within each suite.
 
 ---
 
+---
+
+### `shardGrepEnabled`
+
+<p><small>| OPTIONAL | BOOLEAN |</small></p>
+
+When sharding is configured and used in conjunction with `grep`/`grepInvert`, it may happen that some spec files are allocated to VMs, just to be skipped by Playwright in accordance with the `grep`/`grepInvert` filters, thus wasting VM allocations.
```
Comment (LoryApiFortress) [inline docs/web-apps/automated-testing/playwright/yaml.md:723]:
```suggestion
When sharding is configured and used in conjunction with `grep`/`grepInvert`, some spec files may be allocated to VMs, just to be skipped by Playwright in accordance with the `grep`/`grepInvert` filters, thus wasting VM allocations.
```

File: docs/web-apps/automated-testing/playwright/yaml.md [hunk 2]
```diff
@@ -714,6 +714,34 @@ The `numShards` and `shard` properties are mutually exclusive within each suite.
...
+
+With `shardGrepEnabled: true`, saucectl will ensure that every spec to be allocated contains at least one test matching the `grep`/`grepInvert` filters.
+
+:::tip
+Filtering relies on the value that are set for `grep` and `grepInvert`.
```
Comment (LoryApiFortress) [inline docs/web-apps/automated-testing/playwright/yaml.md:728]:
```suggestion
Filtering relies on the value set for `grep` and `grepInvert`.
```

**Review by LoryApiFortress — APPROVED** [review]
@FriggaHel I left some suggestions. Thank you!

**Review by alexplischke — APPROVED** [review]
LGTM once Lory's suggestions have been applied 👍

**Pull request merged**